server/services/applicationService.ts

```diff
@@ -108,7 +108,13 @@ export default class ApplicationService {
       delete applicationData['funding-information'].identification
       delete applicationData['funding-information']['alternative-identification']
     }
-
+    if (applicationData['equality-and-diversity-monitoring']?.['will-answer-equality-questions'].willAnswer === 'no') {
+      Object.keys(applicationData['equality-and-diversity-monitoring']).forEach(key => {
```

Comment:
would it be simpler here just to reassign? i.e. 

`applicationData['equality-and-diversity-monitoring'] = {'will-answer-equality-questions: { willAnswer: 'no} }`

Reply:
I thought about that but then worried that the structure of `will-answer-equality-questions` might change in the future, and we shouldn't be hardcoding it here? like, the application service shouldn't know or set the shape of page data ideally?